iquip/apps/result_explorer.py

```diff
@@ -0,0 +1,119 @@
+
+    """
+
+    def __init__(self, parent: Optional[QWidget] = None):
+        """Extended."""
+        super().__init__(parent=parent)
+        # widgets
+        self.ridList = QListWidget(self)
+        self.resultInfoList = QListWidget(self)
+        self.reloadButton = QPushButton("Reload", self)
+        self.openButton = QPushButton("Visualize", self)
+        # layout
+        layout = QVBoxLayout(self)
+        layout.addWidget(self.ridList)
+        layout.addWidget(self.resultInfoList)
+        layout.addWidget(self.reloadButton)
+        layout.addWidget(self.openButton)
+        self.setLayout(layout)
+
+
+class _RidListThread(QThread):
+    """QThread for obtaining the RID list from the proxy server.
+    
+    Signals:
+        fetched(ridList):
+          The RID list is fetched.
+    """
+
+    fetched = pyqtSignal(list)
+
+    def __init__(self, callback: Callable[[List[str]], None], parent: Optional[QObject] = None):
+        """Extended.
+        
+        Args:
+            callback: The callback method called after this thread is finished.
+        """
+        super().__init__(parent=parent)
+        self.fetched.connect(callback, type=Qt.QueuedConnection)
+
+    def run(self):
+        """Overridden.
+        
+        Fetches the RID list from the proxy server.
+
+        After finished, the fetched signal is emitted.
+        """
+        try:
+            response = requests.get("http://127.0.0.1:8000/result/", timeout=10)
+            response.raise_for_status()
+            ridList = response.json()
+        except requests.exceptions.RequestException:
+            logger.exception("Failed to fetch the RID list.")
+            return
+        self.fetched.emit(ridList)
+
+
+class ResultExplorerApp(qiwis.BaseApp):
+    """App for showing the RID list and a specific h5 result.
+    
+    Attributes:
+        explorerFrame: The frame that shows the RID list and a specific h5 result.
+        ridListThread: The most recently executed _RidListThread instance.
+    """
+
+    def __init__(self, name: str, parent: Optional[QObject] = None):
+        """Extended."""
+        super().__init__(name, parent=parent)
+        self.ridListThread: Optional[_RidListThread] = None
+        self.explorerFrame = ResultExplorerFrame()
+        self.loadRidList()
+        # connect signals to slots
+        self.explorerFrame.reloadButton.clicked.connect(self.loadRidList)
+
+    @pyqtSlot()
+    def loadRidList(self):
+        """Loads the RID list in self.explorerFrame.ridList."""
+        self.ridListThread = _RidListThread(self._addRid, self)
+        self.ridListThread.start()
+
+    def _addRid(self, ridList: List[str]):
+        """Clears the original RID list and adds the RIDs into self.explorerFrame.ridList.
+        
+        Args:
+            ridList: The fetched RID list.
+        """
+        for _ in range(self.explorerFrame.ridList.count()):
+            item = self.explorerFrame.ridList.takeItem(0)
+            del item
```

Comment:
Is it a bad idea to write theses in a single line, not using a variable `item`?

Reply:
Oh, it looks better😮 I will apply it.

Reply:
I realized that writing them in a single line is impossible.

It causes the following error.

![image](https://github.com/snu-quiqcl/iquip/assets/76851886/1b01f080-0918-4c08-8dab-1dfa8f35dc2c)